### DISCUSSION TITLE
Is it possible to skip weekend on time series chart or line chart?

### DISCUSSION THREAD

**Discussion by dongr0510** [apache/superset#18291] [Q&A / Help]
Hi guys,

I have this issue when I want to show some stock index by using superset; 

Since stock market only opens several hours on weekdays, so I hope there is only Stock trading time on x-axis.

my database format
`
StockID: 'AAPL' 
Price: 130
TradingDate '2021-04-28 09:30:00'
`
Currently, there is only market trading time on my TradingDate column. Superset requires a date format on x-axis for either Line chart or Time-Series chart. And its template will automatically show every hour and every day.

**Is it possible to remove those non trading time? or is it possible to show the time that only exists in the dataset?**

This issue can be reproduced by using any stock market data; it would be better if you have hour/minute level data.

![1619493466365.png](https://user-images.githubusercontent.com/52690436/116179907-a7d36680-a74a-11eb-9bb6-d4ae3386ff3a.png)

**Comment by srinify** [comment]
To unblock you a bit @dongr0510 your best tbh is probably to adjust this at the data level. Few options:

- Add database transforms via views or a tool like dbt to explicitly add 0's for weekends / non trading hours.
- Use the semantic layer in Superset to augment all queries from Charts -> Database such that these values are zero'd out: https://superset.apache.org/docs/creating-charts-dashboards/first-dashboard#superset-semantic-layer 

To really solve this problem longer term, we'd need to add either a separate chart type focused on these types of datasets or add options to the Customization tab that let users of Explore hide x-axis ranges / specific things like weekends easily.

**Comment by dongr0510** [comment]
@srinify Thanks for replying!

This is not a serious problem if I just want to show day level stock market data, since each week only has 2 non trading days. But for hour level stock market data, most of the time are non trading time; since each week has 168 hours, and trading hours is roughly 30% of time.   

To my understanding, those two options are trying to maunally add non trading hour data, and make it 0; but I still cannot remove those non trading hour on the x-axis? 

And I also found this issue on github:
https://github.com/apache/superset/issues/10828
I guess this new feature might solve my issue. If we could have a numeric or categorical line chart, then I can transform TradingDate to the String format; If I could have String format 'TradingDate', then the x-axis might only show non trading hour.

Hope we will have a separate chart or a new hiding x-axis range feature! Thanks!

**Reply by DcPublished** [reply]
I’m very late to this discussion, but… As a Data Engineer withe 30 years of experience, my strong suggest is NEVER replace nulls with 0’s.  This WILL screw up derived metrics like moving averages and linear regressions.   

A NULL means that there is no data.   A “0” is a valid value.  I.e. it is data.   If the value for 5 days a week is always “100” and  for two days a week it will be “0”, the moving average will be less than 100.   E.g. instead of 500/5 the average will be 500/7.    

The correct solution is to exclude weekends from any calculations and from any charts.   That is very difficult when dealing with intraday data.

Dan.
